### PR TITLE
Feature/album favoriting

### LINKED ISSUE #5: As a user, I can favorite an album.
- [x] Render - favorite icon
- [x] Behavior - onClick add fill, POST unique pair to DB
- [x] Behavior - toggle fill on even clicks, DELETE pair from DB
- [x] Behavior - on toggle, array splice() or filter()

### LINKED ISSUE #10: Add "unfavoriting" functionality

### LINKED ISSUE #11: Add favoriting functionality

### PR DESCRIPTION
**Pull request by monshan**
# Selector FE PR

## What does this merge do? Check all that apply.
- [x] Adds a new feature
- [ ] Improves an existing feature
- [ ] Fixes a bug
- [x] Cleans up (logs, tests, etc.)

## Summary of changes

- Several helper function for the favoriting and unfavoriting feature are defined on the App component (highest level)
  - New state on App that holds titles for favorites
  - Adds new favorite to state
  - Removes old favorite from state
  - Toggling that adds or removes based on the elements current existence in state
  - Function that returns a boolean of if state included the element in question or not
- Passes down relevant helpers to the Details component to allow for favoriting and unfavoriting

## Notes for reviewer (any known bugs, areas of interest, etc.)

- Several attempts were made to introduce this feature on both the Details component and AlbumCard component
  - However, without the param being defined on the GraphQL DB and user definitions on DB the team decided against including them
- The primary purpose of this branch has thus been refined to the following User Story:
  - As a User, while on the Detailed view page I can observe a favoriting button that will fill or unfill based on if it is a favorite or not and appear accurately on <code>/your-favorites</code> address on the app

## Next Steps (Future Iterations)

- Add <code>data-cy</code> labels to all relevent elements
- Write testing for the User Flow / Feature

### CODE REVIEW

File: src/Components/App/App.js [hunk 1]
```diff
@@ -25,6 +25,40 @@ function App() {
     return setFavorites([...favorites.filter(album => album.title !== toRemove)]);
   }
 
+  const determineFav = () => {
+    if (isFav) {
```
Comment (rdtho2525) [inline src/Components/App/App.js:29]: This function should return the `FavoriteIcon` nested in the `Tooltip` component in order to toggle the label along with the icon.

**Comment by rdtho2525** [comment]
+ Favoriting capability operates as expected from the Details view
+ `Tooltip` label does not toggle in sync with the favorite icon (see review)
+ Future iteration: adding "unfavoriting" capability in the Favorites view

**Review by rdtho2525 — CHANGES_REQUESTED** [review]
+ Favoriting capability operates as expected from the Details view
+ `Tooltip` label does not toggle in sync with the favorite icon (see review)
+ Future iteration: adding "unfavoriting" capability in the Favorites view

**Comment by monshan** [comment]
> * Favoriting capability operates as expected from the Details view
> * `Tooltip` label does not toggle in sync with the favorite icon (see review)
> * Future iteration: adding "unfavoriting" capability in the Favorites view

ToolTip rendering has been changed!

**Pull request merged**